src/managed_scip.jl

```diff
@@ -0,0 +1,74 @@
+# holds pointers to SCIP data and takes care of memory mgmt
+mutable struct ManagedSCIP
+    scip::Ref{Ptr{SCIP_}}
+    vars::Vector{Ref{Ptr{SCIP_VAR}}}
+    conss::Vector{Ref{Ptr{SCIP_CONS}}}
+
+    function ManagedSCIP()
+        scip = Ref{Ptr{SCIP_}}()
+        @SC SCIPcreate(scip)
+        @assert scip[] != C_NULL
+        @SC SCIPincludeDefaultPlugins(scip[])
+        @SC SCIP.SCIPcreateProbBasic(scip[], "")
+
+        mscip = new(scip, [], [])
+        finalizer(free_scip, mscip)
+    end
+end
+
+"Release references and free memory"
+function free_scip(mscip::ManagedSCIP)
+    # avoid double-free
+    if mscip.scip[] != C_NULL
+        for cons in mscip.conss
+            @SC SCIPreleaseCons(mscip.scip[], cons)
+        end
+        for var in mscip.vars
+            @SC SCIPreleaseVar(mscip.scip[], var)
+        end
+        @SC SCIPfree(mscip.scip)
+    end
+    @assert mscip.scip[] == C_NULL
+end
+
+"Returns pointer to SCIP instance"
+get_scip(mscip::ManagedSCIP) = mscip.scip[]
```

Comment:
Naming: usually we say `X` instead of `get_X`, e.g., `JuMP.value` instead of `JuMP.get_value`.

Reply:
Hm, yes. I wondered that if I added a function `var()`, it might conflict with local variables of the same name?

Reply:
Am using `scip`, `var` and `cons` now.